File: Übungsblatt 4/code/golang/main.go
package main

import (
	"fmt"
	"time"
	"regexp"
)

var results [20]bool
var t [20]time.Duration

func main() {
	pattern := "(.*?,){13}z"
	r,_ := regexp.Compile(pattern)
	for i := 0; i < 20; i++ {
		go func(i int) {
			w := "1,2,3,4,5,6,7,8,9,10,11,12"
			for j:= 0; j < i; j++ {
				w += ",1"
			}
			start := time.Now()
			result := r.MatchString(w)
			tm := time.Now()
			elapsed := tm.Sub(start)
			results[i] = result
			t[i] = elapsed
		}(i)
	}

	var avg int64
	avg = 0;
	fmt.Println("Index | Result | Duration")
	for i:= 0; i < 20; i++ {
		fmt.Printf("%v    ",i)
		if i < 10 {
			fmt.Printf(" ")
		}
		fmt.Printf("| %v  ", results[i])
		if results[i] == true {
			fmt.Printf(" ")
		}
		fmt.Printf("| %v\n", t[i])
		avg += t[i].Microseconds()
	}

	avg /= 20

	fmt.Printf("Average Time: %v\n", avg)
}

File: Übungsblatt 4/code/java/Main.java
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class Main {

    public static void main(String[] args) {
        String patternString = "(.*?,){13}z";
        Pattern pattern;
        boolean[] results = new boolean[20];
        long[] time = new long[20];
        
        pattern = Pattern.compile(patternString);
        for(int i = 1; i <= 20; i++) {
            String text = "1,2,3,4,5,6,7,8,9,10,11,12";
            for(int j = 0; j < i; j++) {
                text += ",1";
            }
            long start = System.nanoTime();
            boolean result = pattern.matcher(text).matches();
            long end = System.nanoTime();
            long elapsed = (end - start) / 1000000;
            results[i-1] = result;
            time[i-1] = elapsed;
        }

        long avg = 0;
        System.out.println("Index | Result | Duration (ms)");
        for(int i = 01; i <= 20; i++) {
            String text = i + "    ";
            if(i < 10) text += " ";
            text += "| " + results[i-1] + "  ";
            if(results[i-1] == true) text += " ";
            text += "| " + time[i-1];
            System.out.println(text);
            avg += time[i-1];
        }

        avg /= 20;

        System.out.println("Average Time (ms); " + avg);
    }
}